cluster-autoscaler/core/scaledown/planner/planner_test.go

```diff
@@ -346,10 +409,34 @@ func generateReplicaSets() []*appsv1.ReplicaSet {
 	}
 }
 
+func generateReplicaSetWithReplicas(name string, specReplicas, statusReplicas int32, labels map[string]string) []*appsv1.ReplicaSet {
```

Comment:
Since you just introduced `ControllerReplicasCalculator` interface, wouldn't it be simpler to just mock it?